python/cudf/cudf/_libxx/column.pyx

```diff
@@ -293,15 +293,18 @@ cdef class Column:
         return self._view(libcudf_types.UNKNOWN_NULL_COUNT).null_count()
 
     cdef mutable_column_view mutable_view(self) except *:
+        col_offset = self.offset
         if is_categorical_dtype(self.dtype):
             col = self.codes
+            # codes is already constructed with offset
+            col_offset = 0
```

Comment:
In what we hand to libcudf we should still use base allocations and let it handle the offset for us:
```suggestion
            col = self.base_children[0]
```